### PR TITLE
fetch heap & nonHeap memory metrics together

### PR DESCRIPTION
**Pull request by vudayani-vmw**

### CODE REVIEW

File: headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverJMX.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ public SpringProcessMemoryMetricsLiveData refreshMemoryMetrics(SpringProcessLive
 				log.info("retrieve live data from: " + jmxURL);
 				SpringProcessMemoryMetricsLiveData liveData = springJMXConnector.retrieveLiveMemoryMetricsData(getProcessType(), jmxConnection, processID, processName, currentData, metricName, tags);
 				
-				if (liveData != null && liveData.getMemoryMetrics() != null && liveData.getMemoryMetrics().length > 0) {
+				if (liveData != null && liveData.getHeapMemoryMetrics() != null && liveData.getHeapMemoryMetrics().length > 0 && liveData.getNonHeapMemoryMetrics().length > 0) {
```
Comment (BoykoAlex) [inline headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverJMX.java:162]: Minor comment. If we are checking liveData.getHeapMemoryMetrics() != null we should do the same for liveData.getNonHeapMemoryMetrics(). Or just don't check for 'null' in both cases

**Comment by BoykoAlex** [comment]
and CLA... @vudayani-vmw githun still thinks CLA is not signed... Does the link to sign the CLA shows somewhere in the PR for you?

File: headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverHttp.java [hunk 2]
```diff
@@ -149,7 +149,7 @@ public SpringProcessMemoryMetricsLiveData refreshMemoryMetrics(SpringProcessLive
                 this.processName = liveData.getProcessName();
             }
 
-            if (liveData != null && liveData.getMemoryMetrics() != null && liveData.getMemoryMetrics().length > 0) {
+            if (liveData != null && liveData.getHeapMemoryMetrics() != null && liveData.getHeapMemoryMetrics().length > 0 && liveData.getNonHeapMemoryMetrics().length > 0) {
```
[the remark this answers is not in the document]
Comment (vudayani-vmw) [reply headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorOverHttp.java]: Added null check for `liveData.getNonHeapMemoryMetrics()`

**Comment by BoykoAlex** [comment]
@vudayani-vmw All looks good. Any chance you can rebase your changes on the latest from the **main** branch?

I had issues rebasing on the latest from **main**. Looks like you've been doing merges rather than rebases and hence you commits are sandwiched together with commits from other folks into **main** branch which makes rebase difficult.

**Comment by BoykoAlex** [comment]
@vudayani-vmw I wasn't able to do a rebase properly. However, it is not too many changes hence I factored them out into a separate PR #874. If it looks good to you (i.e. I didn't miss anything) I'll merge it instead :-)

**Comment by vudayani-vmw** [comment]
@BoykoAlex Thanks for doing this. I will keep this point in mind for the next time. 
The PR https://github.com/spring-projects/sts4/pull/874 looks good to me :-)

**Comment by BoykoAlex** [comment]
Closed via #874

**Pull request closed without merging**